lock-impl/build.gradle

```diff
@@ -1,6 +1,7 @@
 apply from: "../gradle/shared.gradle"
 
 dependencies {
+  api project(":atlasdb-commons-api")
```

Comment:
There's probably a reason we need this, but it's not immediately obvious to me why this is necessary over and above, e.g. `timelock-api`.

This really should be in the description.

Reply:
We moved `TimestampLeaseName` to `:atlasdb-commons-api`, which is used in `lock-impl`. Our gradle plugin orchestration requires all dependencies to be explicitly declared.